qos-monitor/src/main/java/eu/arrowhead/core/qos/dto/event/EventDTOConverter.java

```diff
@@ -0,0 +1,195 @@
+		mapper.configure(SerializationFeature.INDENT_OUTPUT, true);
+	}
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------
+	public static InterruptedMonitoringMeasurementEventDTO convertToInteruptedMonitoringMeasurementEvent(final EventDTO event) {
+		logger.debug("convertToInteruptedMonitoringMeasurementEvent started...");
+
+		Assert.notNull(event, "Event is null");
+		Assert.isTrue(!Utilities.isEmpty(event.getPayload()), "Event payload is empty");
+		Assert.isTrue(!Utilities.isEmpty(event.getTimeStamp()), "Event timeStamp is empty");
+
+		final InterruptedMonitoringMeasurementEventDTO validEvent = new InterruptedMonitoringMeasurementEventDTO();
+		validEvent.setMetadata(event.getMetaData());
+		validEvent.setPayload(event.getPayload());
+		try {
+			validEvent.setTimeStamp(Utilities.parseUTCStringToLocalZonedDateTime(event.getTimeStamp()));
+		} catch (final DateTimeParseException ex) {
+
+			throw new InvalidParameterException("Event timestap has invalid format.", ex);
+		}
+
+		validateInterruptedMonitoringMeasurementEventDTOFields(validEvent);
+
+		return validEvent;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public static FinishedMonitoringMeasurementEventDTO convertToFinishedMonitoringMeasurementEvent(final EventDTO event) {
+		logger.debug("convertToFinishedMonitoringMeasurementEvent started...");
+
+		Assert.notNull(event, "Event is null");
+		Assert.isTrue(!Utilities.isEmpty(event.getPayload()), "Event payload is empty");
+		Assert.isTrue(!Utilities.isEmpty(event.getTimeStamp()), "Event timeStamp is empty");
+
+		final FinishedMonitoringMeasurementEventDTO validEvent = new FinishedMonitoringMeasurementEventDTO();
+		validEvent.setMetadata(event.getMetaData());
+		validEvent.setPayload(convertToIcmpPingResponse(event.getPayload()));
+		try {
+			validEvent.setTimeStamp(Utilities.parseUTCStringToLocalZonedDateTime(event.getTimeStamp()));
+		} catch (final DateTimeParseException ex) {
+
+			throw new InvalidParameterException("Event timestap has invalid format.", ex);
+		}
+
+		validateFinishedMonitoringMeasurementEventDTOFields(validEvent);
+
+		return validEvent;
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	public static List<IcmpPingResponseDTO> convertToIcmpPingResponse(final String payload) {
+		logger.debug("convertToIcmpPingResponse started...");
+
+		Assert.notNull(payload, "Payload is null");
+		Assert.isTrue(!Utilities.isEmpty(payload), "Payload is empty");
+
+		try {
+			final List<IcmpPingResponseDTO> validResponse = mapper.readValue(payload, new TypeReference<List<IcmpPingResponseDTO>>(){});
+
+			return validResponse;
+
+		} catch (final IOException e) {
+
+			throw new InvalidParameterException("Invalid IcmpPingResponse");
```

Comment:
please log the original exception for debug

Reply:
fixed